src/completion/provider/provideCompletionItems.ts

```diff
@@ -0,0 +1,92 @@
+import { resolveConfigPluginFunctionWithInfo } from '@expo/config-plugins/build/utils/plugin-resolver';
+import JsonFile from '@expo/json-file';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { Config, getConfiguration } from './configuration/getConfiguration';
+import { createNodeModuleItem, createPathCompletionItem } from './createCompletionItem';
+import { Context, createContext, ResolveType } from './createContext';
+import { getChildrenOfPath, getPathOfFolderToLookupFiles } from './fileUtils';
+
+export async function provideCompletionItems(
+  document: vscode.TextDocument,
+  position: vscode.Position
+): Promise<vscode.CompletionItem[]> {
+  const context = createContext(document, position);
+
+  if (!context) {
+    return [];
+  }
+
+  const config = await getConfiguration(document.uri);
+
+  return provideAsync(context, config);
+}
+
+/**
+ * Read the package.json and get all of the dependencies that have a valid config plugin.
+ *
+ * @param packageJsonPath
+ * @returns
```

Comment:
Maybe just rely on TS's inferred or explicit returns might be better than just `@returns`?